pkg/pillar/cmd/zedrouter/server.go

```diff
@@ -48,6 +65,15 @@ func createServer4(ctx *zedrouterContext, bridgeIP string, bridgeName string) er
 	mux.Handle("/eve/v1/external_ipv4", ipHandler)
 	hostnameHandler := &hostnameHandler{ctx: ctx}
 	mux.Handle("/eve/v1/hostname", hostnameHandler)
+	metadataHandler := &metadataHandler{ctx: ctx}
+	mux.Handle("/latest/meta-data/", metadataHandler)
+	mux.Handle("/2009-04-04/meta-data/", metadataHandler)
+	openstackHandler := &openstackHandler{ctx: ctx}
+	mux.Handle("/openstack", openstackHandler)
+	mux.Handle("/openstack/", openstackHandler)
```

Comment:
What do we need to offer at the openstack API endpoint? Isn't the AWS schema sufficient?

Reply:
We can use both of them, but cloud-init will do some checks inside before calling of metadata endpoints.
For AWS: https://github.com/canonical/cloud-init/blob/a10838a621f2eef3f97959ae8ffb80bc2cd3b5fd/cloudinit/sources/DataSourceEc2.py#L645-L725
For OpenStack: https://github.com/canonical/cloud-init/blob/a10838a621f2eef3f97959ae8ffb80bc2cd3b5fd/cloudinit/sources/DataSourceOpenStack.py#L233-L246